xdsl/dialects/builtin.py

```diff
@@ -628,49 +635,45 @@ def shape_is_complete(self) -> bool:
         return n == len(self.data.data)
 
     @staticmethod
-    def create_dense_index(
-            type: VectorOrTensorOf[IndexType],
-            data: List[int | IntegerAttr[IndexType]]
```

Comment:
I guess this removal is in the same vain as the whole `build should not take ops` argument? To me, this does not quite feel like a lint pass, but changing "functionality" in terms of pyright issues for users.

Reply:
There are actually several of these in this PR...

Reply:
Yeah that's my b, the PR was originally going to be just this but then I got carried away cleaning up the other errors. This is the main change in this PR for sure. The thought behind it is that it's actually never used with the attribute constructor, and that it makes both the type checking harder and the code less efficient to check for isinstance everywhere.

Reply:
The support for list of `IntegerAttr` is back, although still not by mixing them